src/shared/devnode-acl.c

```diff
@@ -94,68 +92,47 @@ int devnode_acl(const char *path,
 
                 r = acl_find_uid(acl, new_uid, &entry);
                 if (r < 0)
-                        goto finish;
+                        return r;
 
                 if (r == 0) {
-                        if (acl_create_entry(&acl, &entry) < 0) {
-                                r = -errno;
-                                goto finish;
-                        }
+                        if (acl_create_entry(&acl, &entry) < 0)
+                                return -errno;
 
                         if (acl_set_tag_type(entry, ACL_USER) < 0 ||
-                            acl_set_qualifier(entry, &new_uid) < 0) {
-                                r = -errno;
-                                goto finish;
-                        }
+                            acl_set_qualifier(entry, &new_uid) < 0)
+                                return -errno;
                 }
 
-                if (acl_get_permset(entry, &permset) < 0) {
-                        r = -errno;
-                        goto finish;
-                }
+                if (acl_get_permset(entry, &permset) < 0)
+                        return -errno;
 
                 rd = acl_get_perm(permset, ACL_READ);
-                if (rd < 0) {
-                        r = -errno;
-                        goto finish;
-                }
+                if (rd < 0)
+                        return -errno;
 
                 wt = acl_get_perm(permset, ACL_WRITE);
-                if (wt < 0) {
-                        r = -errno;
-                        goto finish;
-                }
+                if (wt < 0)
+                        return -errno;
 
                 if (!rd || !wt) {
 
-                        if (acl_add_perm(permset, ACL_READ|ACL_WRITE) < 0) {
-                                r = -errno;
-                                goto finish;
-                        }
+                        if (acl_add_perm(permset, ACL_READ|ACL_WRITE) < 0)
+                                return -errno;
 
                         changed = true;
                 }
         }
 
         if (!changed)
-                goto finish;
-
-        if (acl_calc_mask(&acl) < 0) {
-                r = -errno;
-                goto finish;
-        }
-
-        if (acl_set_file(path, ACL_TYPE_ACCESS, acl) < 0) {
-                r = -errno;
-                goto finish;
-        }
+                return r;
```

Comment:
i tink it would be better to write `return 0` here, after all we want to return success here, and the return code of some earlier call doesn't matter really at this point.

Reply:
Done. I've also proposed a second commit that reverses that condition, as it looks slightly better to me.